### PR TITLE
Instanced sprites entry point

### LINKED ISSUE #192: Sprite instancing
Sprite rendering has an important use case of tiled maps, e.g. https://github.com/gfx-rs/gfx/tree/pre-ll/examples/ubo_tilemap

We support instancing for meshes, and we should group all the tiles using the same texture (but different texel ranges) as well for instancing.

cc @eddyb @vitvakatu

### PR DESCRIPTION
**Pull request by kvark**
Fixes #192
TODO: instanced tile map example (in follow-ups)

### CODE REVIEW

File: src/render/mod.rs [hunk 1]
```diff
@@ -945,15 +945,18 @@ impl Renderer {
...
+                        Some(ref map) => map.uv_range(),
+                        None => [0.0; 4],
+                    };
+                    if let Some(ref key) = gpu_data.instance_cache_key {
                         let vec = self.instance_cache.entry(key.clone()).or_insert((
                             InstanceData {
                                 slice: gpu_data.slice.clone(),
                                 vertices: gpu_data.vertices.clone(),
-                                pso_data: pso_data.clone(),
+                                pso_data: PsoData::Basic { color, map, param0 },
```
Comment (vitvakatu) [inline src/render/mod.rs]: I can't understand this change, but okay.

Comment (kvark) [reply src/render/mod.rs]: it's basically `pso_data` that has been destructured, so we compose it again
I'll add a comment

**Comment by kvark** [comment]
@vitvakatu I ended up refactoring the instance data a bit

**Comment by kvark** [comment]
bors r=vitvakatu

**Pull request merged**